### PR TITLE
Adds readonly property to PillBadge and Modal button

### PR DESCRIPTION
**Pull request by manivoxel51**
## What changes are proposed in this pull request?

- adds a readonly property to PillBadge and ModalView - needed for enforcing permissioning in teams
- adds an optional tooltipTitle property so we can show things like "you don't have sufficient permissions." in teams context. @imanjra lmk if there is a better way and I'll followup with improvements.

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced a `disabled` property to the `ModalButtonView` for enhanced button control.
	- Added `readOnly` and `tooltipTitle` properties to the `PillBadge` component, allowing it to render in a non-interactive state and provide additional context.
	- Updated the `PillBadgeView` to reflect the new `readOnly` prop for better schema integration.
	- Added a `tooltipTitle` property to the `DropdownView` for enhanced user guidance.

- **Improvements**
	- Enhanced button interactivity based on tagging functionality.
	- Improved usability of the `PillBadge` component by restricting user input when in read-only mode.
	- Adjusted the default cursor style in the `ImageView` component to provide clearer user interaction feedback.

### CODE REVIEW

File: app/packages/components/src/components/PillBadge/PillBadge.tsx [hunk 1]
```diff
@@ -90,6 +92,7 @@ const PillBadge = ({
       ) : (
         <FormControl fullWidth>
           <Chip
+            disabled={readOnly}
```
[the remark this answers is not in the document]
Comment (manivoxel51) [reply app/packages/components/src/components/PillBadge/PillBadge.tsx:95]: fair point

**Review by lanzhenw — APPROVED** [review]

File: app/packages/components/src/components/PillBadge/PillBadge.tsx [hunk 2]
```diff
@@ -10,12 +10,16 @@ const PillBadge = ({
   variant = "filled",
   showIcon = true,
   operator,
+  readOnly = false,
```
Comment (Br2850) [inline app/packages/components/src/components/PillBadge/PillBadge.tsx:13]: Why is this needed for PillBadge?

To set a readOnly PillBadge view you simply pass down a singular string. If only a string is given to this component (with or without a color) it becomes unselectable and is no longer a writeable component

**Review by Br2850 — CHANGES_REQUESTED** [review]
ModalButtonView doesn't have a disabled field at the parent level so that makes sense

But I'm a bit confused as to why the changes to PillBadge & PillBadgeView are needed? If you just pass in a string, the component is automatically read-only

Here's an example: 
```
schema = {"text": "Disabled", "color": "secondary"}
badge = types.PillBadgeView(**schema)
```

File: app/packages/components/src/components/PillBadge/PillBadge.tsx (hunk 2 above)
Comment (manivoxel51) [reply app/packages/components/src/components/PillBadge/PillBadge.tsx:13]: if we simply use 
```
schema = {"text": "Disabled", "color": "secondary"}
badge = types.PillBadgeView(**schema)
```

The badge doesn't reflect it is a dropdown select - the downward caret will be missing. disabled / readOnly are legit properties to use in common components. if you really don't like the readOnly prop, happy to huddle and resolve!

Comment (Br2850) [reply app/packages/components/src/components/PillBadge/PillBadge.tsx:13]: But isn't that the point of readOnly? If I should only be reading the state of something, I don't understand why they would need to see the dropdown caret?

Having a dropdown caret implies that they can change the state of the pill, but we don't want that

So we should just show the pill and whatever the current status is without the ability to change it - which is just the string & a color choice

**Review by Br2850 — APPROVED** [review]
After demo, looks good, this gives us options if we need it

**Pull request merged**